x-pack/plugins/infra/public/pages/logs/stream/page_toolbar.tsx

```diff
@@ -57,7 +58,7 @@ export const LogsToolbar = () => {
           <QueryStringInput
             disableLanguageSwitcher={true}
             iconType="search"
-            indexPatterns={[derivedDataView]}
+            indexPatterns={[derivedDataView as DataView]}
```

Comment:
Isn't this pretty risky? If the `QueryStringInput` requires a full `DataView` instance from now on is there a way to provide one without all the overhead of persisting it?

Reply:
>If the QueryStringInput requires a full DataView instance from now on, is there a way to provide one without all the overhead of persisting it?

`dataViews.create` will create an instance without persisting.

We're working to remove all instances of faked data view instances. Is this happening in the infra code? Perhaps a new issue needs to be created to address this.

Reply:
Yes, that derived data view is "fake". But if data views can now be created without persistence we can replace that. But let's do that separately :+1: